frontend/coprs_frontend/coprs/views/apiv3_ns/apiv3_projects.py

```diff
@@ -181,28 +181,35 @@ def get(self, query, **kwargs):
         return {"items": projects, "meta": paginator.meta}
 
 
-@apiv3_projects_ns.route("/add/<ownername>")
+@apiv3_projects_ns.route("/add/<ownername>/")
 class ProjectAdd(Resource):
     @restx_api_login_required
-    @apiv3_projects_ns.doc(params=ownername_params)
+    @query_to_parameters
+    @apiv3_projects_ns.doc(params=project_params)
     @apiv3_projects_ns.marshal_with(project_model)
     @apiv3_projects_ns.expect(project_add_input_model)
     @apiv3_projects_ns.response(HTTPStatus.OK.value, "Copr project created")
     @apiv3_projects_ns.response(
         HTTPStatus.BAD_REQUEST.value, HTTPStatus.BAD_REQUEST.description
     )
-    def post(self, ownername):
+    def post(self, ownername, exist_ok=False):
         """
         Create new Copr project
         Create new Copr project for ownername with specified data inserted in form.
         """
+        exist_ok = flask.request.args.get("exist_ok") == "True"
         user, group = owner2tuple(ownername)
         data = rename_fields(get_form_compatible_data(preserve=["chroots"]))
-        form_class = forms.CoprFormFactory.create_form_cls(user=user, group=group)
+        form_class = forms.CoprFormFactory.create_form_cls(user=user, group=group,
+                                                           exist_ok=exist_ok)
         set_defaults(data, form_class)
         form = form_class(data, meta={"csrf": False})
 
         if not form.validate_on_submit():
+            if exist_ok:
+                for validator in form.name.validators:
+                    if isinstance(validator, forms.CoprUniqueNameValidator):
+                        return to_dict(validator.copr)
```

Comment:
This deserves some in-line comment :-)

Reply:
Agreed. And thank you for poking it, there was a small bug. I fixed it, and added a comment.